app/assets/javascripts/request_refund.js

```diff
@@ -0,0 +1,34 @@
+$(document).ready(function () {
+  $('.refund-button').click(function () {
+    var uniq_id = $(this).data('id');
+    $('#uniq_id').val(uniq_id);
+  });
+  $('#reason-label').hide();
+  $('input[type="radio"]').click(function () {
+    if ($('#reason-other').is(':checked')) {
+      $('#reason-label').show();
+    } else {
+      $('#reason-label').hide();
+    }
```

Comment:
I think it would be nice to have line `1` to `12`  
```
(document).ready(function () {
    var reason_label = $('#reason-label'), 
        option_state = $('#reason-other').is(':checked'),
        uniq_id = "";

  $('.refund-button').click(function () {
      uniq_id = $(this).data('id');
      $('#uniq_id').val(uniq_id);
  });
  
  reason_label.hide();

  $('input[type="radio"]').click(function () {
   (option_state)? reason_label.show() : reason_label.hide(); 
  });
});

```